### ISSUE TITLE
Json serialization of empty opaque values

### ISSUE THREAD

**Issue by davidgraeff** [eclipse/wakaama#371]
An OPAQUE resource is allowed to be empty, isn't it? If it is, the current code handles this case wrong.

In `json.c` in `prv_serializeValue` line 1122:
```c++
    case LWM2M_TYPE_OPAQUE:
        ...
        res = utils_base64Encode(tlvP->value.asBuffer.buffer, tlvP->value.asBuffer.length, buffer+head, bufferLen - head);
        if (res == 0) return -1;
```
The if condition should be removed. An empty opaque buffer should just result in an empty base64 string.

**Comment by dnav** [comment]
Addressed in #390

**Comment by dnav** [comment]
Fixed by #390.
